tools/lint

```diff
@@ -47,9 +47,12 @@ def run():
     parser.add_option('--full',
                       action='store_true',
                       help='Check some things we typically ignore')
-    parser.add_option('--pep8',
-                      action='store_true',
-                      help='Run the pep8 checker')
+    parser.add_option('--py',
+                      action="store_true",
+                      help='Don\'t run expensive Javascript checks')
+    parser.add_option('--js',
+                      action="store_true",
+                      help='Don\'t run expensive Python checks')
```

Comment:
Yeah, I think we should do it by extension.  I like `--frontend` and `--backend`; if we can get that split right, we could split the linter in Travis to run the backend half in the backend suite and the frontend half in the frontend suite, which would be really nice.  

I think we can do it by just listing the file extensions for each.  Won't be perfect (where does "JSON" go) but should be able to get close enough.